### ISSUE TITLE
[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.NullPointerException when sign in/out Azure account

### ISSUE THREAD

**Issue by codeclan04** [microsoft/azure-tools-for-java#2757]
IntelliJ build version: 2018.3.3 IU-183.5153.38
OS: Mac OS X
JDK: JetBrains s.r.o 1.8.0_152-release
Plugin version: 3.18.0-2018.3
Additional Info: None
Parent component: 
```
com.intellij.openapi.wm.impl.IdeRootPane[,0,22,1440x855,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.NullPointerException
	at com.intellij.openapi.wm.impl.InternalDecorator.setTitleActions(InternalDecorator.java:316)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.setTitleActions(ToolWindowImpl.java:309)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.lambda$addToolbarItems$3(ServerExplorerToolWindowFactory.java:374)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.notifySignOutEventListener(AuthMethodManager.java:106)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.signOut(AuthMethodManager.java:167)
	at com.microsoft.azuretools.ijidea.actions.AzureSignInAction.onAzureSignIn(AzureSignInAction.java:114)
	at com.microsoft.intellij.serviceexplorer.azure.SignInOutAction$1.actionPerformed(SignInOutAction.java:46)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener.actionPerformedAsync(NodeActionListener.java:72)
	at com.microsoft.tooling.msservices.serviceex

**Comment by wezhang** [comment]
@andxu , could you help follow it? Thanks!

**Comment by wezhang** [comment]
@Flanker32

**Comment by Flanker32** [comment]
@codeclan04 Thanks for your report, this seems caused by wrong behavior when create new service principal with device login, this will be fixed in 3.20.0.

**Comment by Flanker32** [comment]
We didn't remove the tool window listener when close a project, fixed in #3431
